circuit/src/queryBuilder/types.ts

```diff
@@ -0,0 +1,82 @@
+import { ethers } from "ethers";
+
+import {
+  AxiomV2Callback,
+  AxiomV2ComputeQuery,
+  AxiomV2DataQuery,
+  AxiomV2FeeData,
+} from "@axiom-crypto/tools";
+
+export {
+  AxiomV2Callback,
+  AxiomV2ComputeQuery,
+  AxiomV2DataQuery,
+  AxiomV2FeeData,
+  DataSubquery,
+  HeaderSubquery,
+  AccountSubquery,
+  StorageSubquery,
+  TxSubquery,
+  ReceiptSubquery,
+  Subquery,
+  SolidityNestedMappingSubquery,
+  BeaconValidatorSubquery,
+  DataSubqueryType,
+  HeaderField,
+  AccountField,
+  TxField,
+  ReceiptField,
+  AxiomV2CircuitConstant,
+  AxiomV2FieldConstant,
+} from "@axiom-crypto/tools";
+
+export interface AxiomV2QueryBuilderConfig {
+  provider: string;
+  chainId?: number | string | BigInt;
+  targetChainId?: number | string | BigInt;
+  version?: string;
+  privateKey?: string;
+  mock?: boolean;
+}
+
+export interface InternalConfig {
+  providerUri: string;
+  chainId: BigInt;
+  targetChainId: BigInt;
+  mock: boolean;
+  provider: ethers.JsonRpcProvider;
+  version: string;
+  signer?: ethers.Wallet;
```

Comment:
remove `signer`: the `circuit` package should not own anything related to signer